src/components/InventoryTable/InventoryList.test.js

```diff
@@ -97,19 +114,17 @@ describe('InventoryList', () => {
           </Provider>
         </MemoryRouter>
       );
-      const wrapper = mount(<Cmp sortBy={sortBy} />);
+      const { rerender } = render(<Cmp sortBy={sortBy} />);
 
-      onRefreshData.mockClear();
-
-      wrapper.setProps({
-        sortBy: {
-          ...sortBy,
-          direction: 'desc',
-        },
-      });
-      wrapper.update();
-
-      expect(onRefreshData).toHaveBeenCalled();
+      rerender(
+        <Cmp
+          sortBy={{
+            ...sortBy,
+            direction: 'desc',
+          }}
+        />
+      );
+      expect(onRefreshData).toHaveBeenCalledTimes(1);
```

Comment:
If the mock has been cleared before the test, wouldn't onRefreshData be expected to have been called twice, once without direction, when it initially loads, and the second time with direction?

Reply:
onRefreshData, in this scenario and given the props, is called only when there is a change to sortBy prop: https://github.com/RedHatInsights/insights-inventory-frontend/blob/master/src/components/InventoryTable/InventoryList.js#L46. This is why on the initial render it's not triggered.